### PR TITLE
add github actions

### PR DESCRIPTION
**Pull request by mcanlas**
This gets us some of the way there but doesn't replicate any of the publishing functionality.

It's a bit annoying to test due to the side-effecting nature of publishing and the branch-specific filter on the job runner.

Feedback/changes welcome!

### CODE REVIEW

**Comment by pathikrit** [comment]
This section is missing:
```
    - stage: publish
      if: branch = master AND type = push AND NOT fork
      jdk: openjdk8
      scala: 2.12.13
      script:
        - sbt ++$TRAVIS_SCALA_VERSION coverage test # test once more to produce coverage reports (https://docs.travis-ci.com/user/build-stages/#data-persistence-between-stages-and-jobs)
        - sbt coverageReport coverageAggregate codacyCoverage makeSite
        - curl -s https://codecov.io/bash | bash
      deploy:
        - provider: pages
          skip_cleanup: true
          token: $GITHUB_TOKEN
          local_dir: target/site
          verbose: true
```

Either we port this to github workflow or we keep the `.travis.yml` file for now in parallel?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Scala CI
+
+on:
+  push:
+    branches: [ master ]
```
Comment (pathikrit) [inline .github/workflows/ci.yml:5]: would this also enable PR merge blockers?

**Comment by mcanlas** [comment]
I was under the (mistaken?) impression that the travis CI didn't work at all now. If it still works lets keep it!

**Comment by pathikrit** [comment]
> I was under the (mistaken?) impression that the travis CI didn't work at all now. If it still works let's keep it!

No it works just fine

**Comment by SethTisue** [comment]
> No it works just fine

I don't think so; look at all the PRs https://github.com/pathikrit/better-files/pulls that don't have CI runs.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mkurz) [reply .github/workflows/ci.yml:5]: What do you mean? This line just says: "when there is a push to the master branch, run this workflow file here"

**Comment by SethTisue** [comment]
@pathikrit what's the path forward here?

**Comment by mkurz** [comment]
Would be nice to get this working and to also get Scala 3 artifacts out asap.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mkurz) [reply .github/workflows/ci.yml:5]: I recommend to also run tests on pull requests, so this section should look like:
```yml
on:
  pull_request:
  push:
    branches: [ master ]
```

**Comment by mkurz** [comment]
I can help out working on a `publish.yml`, shouldn't be really hard actually.

**Comment by pathikrit** [comment]
Ok I moved over to Github actions now. Please review:
https://github.com/pathikrit/better-files/blob/master/.github/workflows/scala.yml

**Pull request closed without merging**